### DISCUSSION TITLE
Failed to use the Library on Arduino Uno + Ethernet Shield + DHT11>2

### DISCUSSION THREAD

**Discussion by MarianTudor82** [dawidchyrzynski/arduino-home-assistant#98] [General]
Hello,

Does anybody manage to connect and transmit the values of more that 2 DHT11 sensors to HA. after I was adding the comand dht3.begin() for the third sensor, the communication to HA failed...

I tried also to use the PubSubClient and Mqtt libraries (without mqttdiscovery) and I got the same issue... Please let me know if you are able to do this test (even with other arduino board).

Thank you!

**Comment by dawidchyrzynski** [comment]
Hi @MarianTudor82,

Arduino Uno has only 2KB of RAM so I suspect you're hitting the resources limit. Handling Ethernet and MQTT is quite expensive. I would suggest migrating to another board/platform (e.g. Arduino Mega, Nano 33 IoT or ESP).

**Reply by MarianTudor82** [reply]
Hello , @dawidchyrzynski !

Thank you for your prompt and useful answer. Please let me know if you already test this scenario with Arduino Mega or if you are able to do it... It will be useful for me to know this before I'll order a few of them for my current project!  Thank you in advance!

**Reply by dawidchyrzynski** [reply]
@MarianTudor82, If you don't have any obvious issue in the codebase then it will work on mega. I successfully use it in my projects where there are 16 switches and 20 triggers implemented on a single device (Controllino Mega - ATmega2560).

**Reply by MarianTudor82** [reply]
Hello @dawidchyrzynski . Fair enough... Thank you for your support and for your work!
Marian
